cloud/auth/auth.go

```diff
@@ -289,25 +290,46 @@ func switchToLastUsedWorkspace(c *config.Context, workspaces []astro.Workspace)
 	return astro.Workspace{}, false, nil
 }
 
-// checkToken requests a users rolebindings and sets the workspace to make sure that token works
-// TODO check orgID is in the token
-func checkToken(c *config.Context, client astro.Client, out io.Writer) error {
-	self, err := client.GetUserInfo()
+// check client status after a successfully login
+func checkUserSession(c *config.Context, client astro.Client, coreClient astrocore.CoreClient, out io.Writer) error {
+	// fetch self user base on token
+	// we set CreateIfNotExist to true so we always create astro user when a successfully login
+	createIfNotExist := true
+	selfResp, err := coreClient.GetSelfUserWithResponse(http_context.Background(), &astrocore.GetSelfUserParams{
+		CreateIfNotExist: &createIfNotExist,
+	})
+	err = astrocore.NormalizeAPIError(selfResp.HTTPResponse, selfResp.Body, err)
 	if err != nil {
 		return err
 	}
-	organizationID := self.AuthenticatedOrganizationID
-
-	if organizationID != "" {
-		err = c.SetContextKey("organization", organizationID)
-		if err != nil {
-			return err
+	activeOrgID := c.Organization
+	// OrganizationId is optional, it may not returned by getSelf api
+	if selfResp.JSON200.OrganizationId != nil {
+		activeOrgID = *selfResp.JSON200.OrganizationId
+	}
+	// fetch all orgs that the user can access
+	orgsResp, err := coreClient.ListOrganizationsWithResponse(http_context.Background())
+	err = astrocore.NormalizeAPIError(orgsResp.HTTPResponse, orgsResp.Body, err)
+	if err != nil {
+		return err
+	}
+	orgs := *orgsResp.JSON200
+	// default to first one in case something crazy happen lol
+	activeOrg := orgs[0]
```

Comment:
do we need to pick the first org? 
It is already set to `activeOrgID := c.Organization` or `*selfResp.JSON200.OrganizationId`

Reply:
1. selfResp.JSON200.OrganizationId is an optional field.
2. c.Organization is the existing org from config but user may have been removed from the org.

This is a fallback scenario that in case 1 and 2 happen, we still log user in with the first org that he can access.

Reply:
good catch, so in this case we just display a message in console and ask user reach out to org owner, does it sound right?

Reply:
Fixed, throw an error with message `no Organization found. Please contact your Astro Organization Owner to be invited to the organization`